src/main/java/local/alfonso/orders/service/CustomerServiceImpl.java

```diff
@@ -0,0 +1,149 @@
+        {
+            throw new EntityNotFoundException("Customer " + name + " not found.");
+        }
+        return customer;
+    }
+
+    @Override
+    public void delete(long id)
+    {
+        if(custrepos.findById(id).isPresent())
+        {
+            custrepos.deleteById(id);
+        } else {
+            throw new EntityNotFoundException(Long.toString(id));
+        }
+    }
+
+    @Transactional
+    @Override
+    public Customers save(Customers customer) {
+        Customers newCustomer = new Customers();
+
+        newCustomer.setCustname(customer.getCustname());
+        newCustomer.setAgent(customer.getAgent());
+        newCustomer.setCustcity(customer.getCustcity());
+        newCustomer.setCustcountry(customer.getCustcountry());
+        newCustomer.setWorkingarea(customer.getWorkingarea());
+        newCustomer.setGrade(customer.getGrade());
+        newCustomer.setOpeningamt(customer.getOpeningamt());
+        newCustomer.setPhone(customer.getPhone());
+        newCustomer.setReceiveamt(customer.getReceiveamt());
+        newCustomer.setPaymentamt(customer.getPaymentamt());
+        newCustomer.setOutstandingamt(customer.getOutstandingamt());
+
+        for (Orders o : customer.getOrders())
+        {
+            newCustomer.getOrders().add(new Orders(o.getOrdamount(), o.getAdvanceamount(), o.getCustomer(), o.getAgent(), o.getOrdersescription()));
+        }
+        return custrepos.save(newCustomer);
+    }
+
+    @Transactional
+    @Override
+    public Customers update(Customers customer, long id)
+    {
+        Customers currentCustomer = custrepos.findById(id).orElseThrow(() -> new EntityNotFoundException(Long.toString(id)));
+
+        if (customer.getOrders().size() > 0)
+        {
+            for (Orders o : customer.getOrders())
+            {
+                currentCustomer.getOrders().add(new Orders(o.getOrdamount(), o.getAdvanceamount(), o.getCustomer(), o.getAgent(), o.getOrdersescription()));
+            }
+        }
+
+        if(customer.getAgent() != null)
+        {
+            currentCustomer.setAgent(customer.getAgent());
+        }
+
+        if(customer.getCustcity() != null)
+        {
+            currentCustomer.setCustcity(customer.getCustcity());
+        }
+
+        if(customer.getCustcountry() != null)
+        {
+            currentCustomer.setCustcountry(customer.getCustcountry());
+        }
+
+        if(customer.getCustname() != null)
+        {
+            currentCustomer.setCustname(customer.getCustname());
+        }
+
+        if(customer.getWorkingarea() != null)
+        {
+            currentCustomer.setWorkingarea(customer.getWorkingarea());
+        }
+
+        if(customer.getGrade() != null)
+        {
+            currentCustomer.setGrade(customer.getGrade());
+        }
+
+        if(customer.getOpeningamt() != currentCustomer.getOpeningamt())
+        {
+            currentCustomer.setOpeningamt(customer.getOpeningamt());
+        }
```

Comment:
As your code is right now, this will not work and throw a `NullPointerException` if the update object's `double` type values are not supplied.

You can fix this by using the wrapper class `Double` instead (uppercase 'D') on your model's variable declarations and the corresponding getters and setters. This is important because wrapper classes like `Double` can contain `null` values, whereas primitive types like `double` cannot.